### ISSUE TITLE
Timeout Issue with Puppeteer Version 16+ When Using Browserless and Docker

### ISSUE THREAD

**Issue by LittleToni** [browserless/chrome#3484]
**Description:**
When I attempt to use browserless in combination with puppeteer (version 16 and above) and docker-compose, I encounter a timeout issue. Interestingly, replacing the local instance of browserless with the cloud service seems to solve the issue, irrespective of the puppeteer version installed.

**Steps to Reproduce:**
1. Set up a project with browserless, puppeteer and docker-compose (https://github.com/LittleToni/browserless-docker-compose-issue-demo).
2. Call `localhost/check` with puppeteer  <= v15 installed.
3. Call `localhost/check` with puppeteer  >= v16 installed.

**Comment by andyMrtnzP** [comment]
Sorry for the delay. I ran the project and realized that the debugger is working at `localhost:3000`, so it seems this is a network error. Do you have a `browserless` entry on your hostfile? I ask because of the `browserless:3000` connection you're making.

Any way, the project worked changing that localhost for my local ip.

**Comment by LittleToni** [comment]
Hey @andyMrtnzP, the connection at ws://browserless:3000 is based on docker compose (https://docs.docker.com/compose/networking/) and is working for puppeteer <= v15. 

> By default Compose sets up a single [network](https://docs.docker.com/engine/reference/commandline/network_create/) for your app. Each container for a service joins the default network and is both reachable by other containers on that network, and discoverable by them at a hostname identical to the container name.

**Comment by andyMrtnzP** [comment]
Ah, I see. I was able to get it work passing the `--project-name` to the compose:

`docker compose --project-name browserless up`

``jsonc

// http://localhost/check
{
  "status": "success",
  "statusCode": 200
}
```

**Comment by joelgriffith** [comment]
Also wanted to raise that you should double check the version of puppeteer your running locally as a version mismatch can also cause connection issues.

**Comment by LittleToni** [comment]
Hi @joelgriffith ,

Thank you for the feedback. I've double-checked the version of puppeteer as you suggested, and I can confirm that there isn't a version mismatch.

To further illustrate the issue, I've updated the demo repository. Now, there are two endpoints – one that works and one that doesn't. This should provide a clearer picture of the problem I'm facing. I would appreciate it if you could take a moment to test the demo yourself and verify the functionality I'm referencing.

Given the new information and updates, I kindly request that you reconsider reopening the issue. I believe that this isn't just a simple configuration or version mismatch problem.

**Comment by andyMrtnzP** [comment]
I'm not able to replicate it this time. Works at first try on my end :/

![image](https://github.com/browserless/chrome/assets/93541085/4ba966c6-5359-45ac-a7da-6b2a7baa5ac2)

**Comment by LittleToni** [comment]
Hi @andyMrtnzP,

Thank you for your patience and for trying to replicate the issue on your end. After further investigation on my side, especially after testing on a different system, I realized that the problem was indeed on my end.

It turns out the root cause of the erroneous behavior on my machine was an outdated Docker image in the cache, which I had downloaded a while ago using the "latest" tag. This image was seemingly not compatible with puppeteer version >= 16. Once I pulled an updated image from Docker Hub, everything worked as expected.

This served as a reminder of how important it is to regularly update Docker images, especially when using the "latest" tag. For future projects, I'll consider using specific tags instead of "latest" to ensure such incidents don't recur and to maintain the stability of my applications.

Thank you once again for your assistance.
